test/fw/ptl/utils/pbs_dshutils.py

```diff
@@ -2070,55 +2070,80 @@ def create_temp_dir(self, hostname=None, suffix='', prefix='PtlPbs',
+                self.chmod(hostname, path=dirname, mode=0o755, runas=asuser)
+
             tmpdir = dirname + tmpdir[4:]
 
         # if temp dir to be created on remote host
         if not self.is_localhost(hostname):
             if asuser is not None:
-                # by default mkstemp creates dir with 0600 permission
+                # by default mkstemp creates dir with 0o700 permission
                 # to create dir as different user first change the dir
-                # permission to 0644 so that other user has read permission
+                # permission to 0o755 so that other user has read permission
                 self.chmod(path=tmpdir, mode=0o755)
                 # copy temp dir created on local host to remote host
                 # as different user
                 self.run_copy(hostname, src=tmpdir, dest=tmpdir, runas=asuser,
-                              recursive=True,
-                              preserve_permission=False, level=level)
+                              recursive=True, uid=uid, gid=gid,
+                              level=level)
+                if mode is not None:
+                    self.chmod(hostname, path=tmpdir, mode=mode, runas=asuser)
+                else:
+                    self.chmod(hostname, path=tmpdir, mode=0o755, runas=asuser)
             else:
+                self.chmod(path=tmpdir, mode=0o755)
                 # copy temp dir created on localhost to remote as current user
                 self.run_copy(hostname, src=tmpdir, dest=tmpdir,
-                              preserve_permission=False, level=level)
+                              level=level, preserve_permission=True,
+                              uid=uid, gid=gid)
             # remove local temp dir
             os.rmdir(tmpdir)
         if asuser is not None:
-            # by default mkdtemp creates dir with 0600 permission
+            # by default mkdtemp creates dir with 0o700 permission
             # to create dir as different user first change the dir
-            # permission to 0644 so that other user has read permission
+            # permission to 0o755 so that other user has read permission
             self.chmod(path=tmpdir, mode=0o755)
             # since we need to create as differnt user than current user
             # create a temp dir just to get temp dir name with absolute path
             tmpdir2 = tempfile.mkdtemp(suffix, prefix, dirname)
             os.rmdir(tmpdir2)
             # copy the orginal temp as new temp dir
             self.run_copy(hostname, src=tmpdir, dest=tmpdir2, runas=asuser,
-                          recursive=True,
-                          preserve_permission=False, level=level)
+                          recursive=True, uid=uid, gid=gid, level=level)
+            if mode is not None:
+                self.chmod(hostname, path=tmpdir2, mode=mode, runas=asuser)
+            else:
+                self.chmod(hostname, path=tmpdir2, mode=0o755, runas=asuser)
```

Comment:
Couldn't you avoid this if statement if you defaulted mode to 0o755?

Reply:
You are right @bhroam.
I'm thinking to just adjust the comment and just not check for 0o700 at all. The reason being, the default permissions would depend on umask value on the system. So assuming its going to be 0o700 may not be right.
I'll also remove these else statements because if the user has not given the mode, let them get what the system provides.
What do you think?

Reply:
I'm not sure.  While using the umask of the system is a very unixy thing it do, it could cause a test to fail on one system and not another due to the different umask.  If we hammer it to the default of 0o755, it will be the same on all systems.  I think I'd agree with you if this wasn't a testing environment.

Reply:
I agree, its probably better to have defined behavior for tests.